enginetest/enginetests.go

```diff
@@ -2206,488 +1971,101 @@ func TestVersionedViewsPrepared(t *testing.T, harness Harness) {
-		ctx.SetCurrentDatabase("")
-
-		TestQueryWithContext(t, ctx, e, "CREATE TABLE mydb.t12 (a INTEGER NOT NULL PRIMARY KEY, "+
-			"b VARCHAR(10) UNIQUE, c varchar(10) UNIQUE)", []sql.Row{{sql.NewOkResult(0)}}, nil, nil)
-
-		db, err := e.Analyzer.Catalog.Database(ctx, "mydb")
-		require.NoError(t, err)
-
-		t12Table, ok, err := db.GetTableInsensitive(ctx, "t12")
-		require.NoError(t, err)
-		require.True(t, ok)
-
-		t9TableIndexable, ok := t12Table.(sql.IndexedTable)
-		require.True(t, ok)
-		t9Indexes, err := t9TableIndexable.GetIndexes(ctx)
-		require.NoError(t, err)
-		uniqueCount := 0
-		for _, index := range t9Indexes {
-			if index.IsUnique() {
-				uniqueCount += 1
-			}
-		}
-
-		// We want two unique indexes to be created with unique names being generated. It is up to the integrator
-		// to decide how empty string indexes are created. Adding in the primary key gives us a result of 3.
-		require.Equal(t, 3, uniqueCount)
-
-		// Validate No Unique Index has an empty Name
-		for _, index := range t9Indexes {
-			require.True(t, index.ID() != "")
-		}
-	})
-	//TODO: Implement "CREATE TABLE otherDb.tableName"
-	t.Run("CREATE TABLE with auto_increment on unique column", func(t *testing.T) {
-		TestQuery(t, harness, e, "CREATE TABLE auto_t1 (i int primary key, j int auto_increment unique)", []sql.Row{{sql.NewOkResult(0)}}, nil)
-
-		db, err := e.Analyzer.Catalog.Database(ctx, "mydb")
-		require.NoError(t, err)
-
-		ctx := NewContext(harness)
-		testTable, ok, err := db.GetTableInsensitive(ctx, "auto_t1")
-		require.NoError(t, err)
-		require.True(t, ok)
-
-		s := sql.Schema{
-			{Name: "i", Type: sql.Int32, Nullable: false, Source: "auto_t1", PrimaryKey: true},
-			{Name: "j", Type: sql.Int32, Nullable: true, Source: "auto_t1", AutoIncrement: true, Extra: "auto_increment"},
-		}
-
-		require.Equal(t, s, testTable.Schema())
-	})
-
-	t.Run("CREATE TABLE with auto_increment on index column", func(t *testing.T) {
-		TestQuery(t, harness, e, "CREATE TABLE auto_t2 (i int primary key, j int auto_increment, index (j))", []sql.Row{{sql.NewOkResult(0)}}, nil)
-
-		db, err := e.Analyzer.Catalog.Database(ctx, "mydb")
-		require.NoError(t, err)
-
-		ctx := NewContext(harness)
-		testTable, ok, err := db.GetTableInsensitive(ctx, "auto_t2")
-		require.NoError(t, err)
-		require.True(t, ok)
-
-		s := sql.Schema{
-			{Name: "i", Type: sql.Int32, Nullable: false, Source: "auto_t2", PrimaryKey: true},
-			{Name: "j", Type: sql.Int32, Nullable: true, Source: "auto_t2", AutoIncrement: true, Extra: "auto_increment"},
-		}
-
-		require.Equal(t, s, testTable.Schema())
-	})
-
-	t.Run("CREATE TABLE with auto_increment on multiple unique columns", func(t *testing.T) {
-		TestQuery(t, harness, e, "CREATE TABLE auto_t3 (i int primary key, j int auto_increment, k int, unique(j,k))", []sql.Row{{sql.NewOkResult(0)}}, nil)
+func TestCreateTable(t *testing.T, harness Harness) {
+	harness.SetSetup("mydb", "mytable", "foo")
+	for _, tt := range CreateTableQueries {
```

Comment:
converted most create tests into WriteQueryTests, a few were not easily transferrable